packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -88,10 +85,6 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
                 props.push(property);
             }
         }
-        // tslint:disable-next-line:forin
-        for (const property of props) {
```

Comment:
Schemas can be contributed dynamically via `setSchema`. In this case, default values should be updated.

Reply:
you are right. i missed the case you described. 

as you can tell from my code, i splitted the original `PreferenceSchemaProvider` into 2 classes: the new `PreferenceSchemaProvider` (which is supposed to provide schemas) and the `DefaultPrefrenceProvider` (which is meant to provide values). I will make it work in the way that both schemas and values can be updated.

Reply:
This change has been reverted.